acstools/acszpt.py

```diff
@@ -0,0 +1,329 @@
+>>> zpt_table = q.fetch()
+>>> print(zpt_table)
+FILTER PHOTPLAM        PHOTFLAM         STmag  VEGAmag  ABmag
+       Angstrom erg / (Angstrom cm2 s) mag(ST)   mag   mag(AB)
+ str6  float64         float64         float64 float64 float64
+------ -------- ---------------------- ------- ------- -------
+ F435W   4329.2              3.148e-19  25.155  25.763  25.665
+ F475W   4746.2              1.827e-19  25.746  26.149  26.056
+ F502N   5023.0              5.259e-18  22.098  22.365  22.285
+ F550M   5581.5               3.99e-19  24.898  24.825  24.856
+ F555W   5360.9              1.963e-19  25.667  25.713  25.713
+ F606W   5922.0              7.811e-20  26.668  26.405  26.498
+ F625W   6312.0              1.188e-19  26.213  25.735  25.904
+ F658N   6584.0               1.97e-18  23.164  22.381  22.763
+ F660N   6599.4              5.156e-18  22.119  21.428  21.714
+ F775W   7693.2              9.954e-20  26.405  25.272  25.667
+ F814W   8045.0              7.046e-20   26.78  25.517  25.944
+F850LP   9033.2               1.52e-19  25.945  24.332  24.858
+ F892N   8914.8              1.502e-18  23.458  21.905    22.4
+
+Retrieve the zeropoint information for the F435W filter on 2016-04-01 for WFC:
+
+>>> from acstools import acszpt
+>>> date = '2016-04-01'
+>>> detector='WFC'
+>>> filt = 'F435W'
+>>> q = acszpt.Query(date=date, detector=detector, filter=filt)
+>>> zpt_table = q.fetch()
+>>> print(zpt_table)
+FILTER PHOTPLAM        PHOTFLAM         STmag  VEGAmag  ABmag
+       Angstrom erg / (Angstrom cm2 s) mag(ST)   mag   mag(AB)
+------ -------- ---------------------- ------- ------- -------
+ F435W   4329.2              3.148e-19  25.155  25.763  25.665
+"""
+
+
+
+
+from astropy.table import Table
+import astropy.units as u
+from collections import OrderedDict
+import datetime as dt
+import requests
+from bs4 import BeautifulSoup
+
+
+__taskname__ = "acszpt"
+__version__ = "1.0"
+__author__ = "Nathan Miles"
+__vdate__ = "22-Jan-2019"
+__all__ = ['acszpt']
+
+
+
+class Query(object):
+    """ Class used to interface with the ACS Zeropoints Calculator API.
+
+    Parameters
+    ----------
+    date : str
+        Input date in the following ISO format, YYYY-MM-DD
+    detector : str
+        One of the three channels on ACS: HRC, SBC, or WFC
+    filt : str, optional
+        One of valid filters for the chosen detector
+            * HRC:
+                 F220W, F250W, F330W,
+                 F344N, F435W, F475W,
+                 F502N, F550M, F555W,
+                 F606W, F625W, F658N, F660N,
+                 F775W, F814W, F850LP, F892N
+
+            * WFC:
+                 F435W, F475W,
+                 F502N, F550M, F555W,
+                 F606W, F625W, F658N, F660N,
+                 F775W, F814W, F850LP, F892N
+            * SBC:
+                F115LP, F122M, F125LP,
+                F140LP, F150LP, F165LP
+
+    Attributes
+    ----------
+    date : str
+        **Attribute** for storing the user specified date.
```

Comment:
No reason to highlight nor mention `**Attribute**`. These are already in the "Attributes" section, should be obvious enough?

Reply:
That's for the RTD pages. For some reason, I could not get autodoc to generate a header over those definitions and so it just looks like a repetition of the keyword args passed to the instance. It correctly parses all public variables and methods, but for some reason will not generate a header like it does for the parameters even though I was following the [numpydoc conventions](https://numpydoc.readthedocs.io/en/latest/format.html#class-docstring) for classes. I ended up removing the Methods section in the class docstring because it was redundant with the methods docstring, in case you're wondering why it's not there.